composer/loggers/neptune_logger.py

```diff
@@ -122,7 +122,7 @@ def neptune_run(self):
             self._neptune_run = Run(
                 project=self._project,
                 api_token=self._api_token,
-                mode=self._mode,
+                mode=self._mode,  # type: ignore
```

Comment:
```suggestion
                mode=self._mode,
```
This change seems spurious

Reply:
when running the pre commit hooks, pyright actually gives lots of errors and warnings now. Originally there was just this one. I think I'm running the correct version of pyright:
```
composer/composer/callbacks/eval_output_logging_callback.py
  composer/composer/callbacks/eval_output_logging_callback.py:111:39 - error: Argument of type "list[str] | None" cannot be assigned to parameter "columns" of type "List[str]" in function "log_table"
    Type "list[str] | None" cannot be assigned to type "List[str]"
      Type "None" cannot be assigned to type "List[str]" (reportGeneralTypeIssues)
  composer/composer/callbacks/eval_output_logging_callback.py:111:64 - error: Argument of type "str | None" cannot be assigned to parameter "name" of type "str" in function "log_table"
    Type "str | None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
composer/composer/utils/checkpoint.py
  composer/composer/utils/checkpoint.py:230:61 - error: Cannot access member "get_local_rank" for type "DeviceMesh"
    Member "get_local_rank" is unknown (reportGeneralTypeIssues)
  composer/composer/utils/checkpoint.py:283:56 - error: Cannot access member "get_group" for type "DeviceMesh"
    Member "get_group" is unknown (reportGeneralTypeIssues)
composer/composer/utils/object_store/sftp_object_store.py
  composer/composer/utils/object_store/sftp_object_store.py:21:12 - warning: Import "paramiko.client" could not be resolved from source (reportMissingModuleSource)
  composer/composer/utils/object_store/sftp_object_store.py:22:10 - warning: Import "paramiko" could not be resolved from source (reportMissingModuleSource)
  composer/composer/utils/object_store/sftp_object_store.py:190:14 - warning: Import "paramiko" could not be resolved from source (reportMissingModuleSource)
composer/tests/utils/object_store/object_store_settings.py
  composer/tests/utils/object_store/object_store_settings.py:45:12 - warning: Import "paramiko" could not be resolved from source (reportMissingModuleSource)
4 errors, 4 warnings, 0 informations

```